From d9a906cf85b819274ce09820a1ea133fadd685e8 Mon Sep 17 00:00:00 2001
From: Valentine Barshak <valentine.barshak@cogentembedded.com>
Date: Thu, 28 Jan 2021 01:55:37 +0300
Subject: [PATCH 05/18] drivers: renesas: rcar: pfc: Update common registers

This uses PFC_POCCTRL1 instead of PFC_IOCTRL31 register name,
making it consistent with the h/w manual, and adds more PFC
POC CTRL registers.

Signed-off-by: Valentine Barshak <valentine.barshak@cogentembedded.com>
---
 drivers/renesas/rcar/pfc/V3M/pfc_init_v3m.c | 2 +-
 drivers/renesas/rcar/pfc/pfc_regs.h         | 3 ++-
 2 files changed, 3 insertions(+), 2 deletions(-)

diff --git a/drivers/renesas/rcar/pfc/V3M/pfc_init_v3m.c b/drivers/renesas/rcar/pfc/V3M/pfc_init_v3m.c
index 6063758..af356b7 100644
--- a/drivers/renesas/rcar/pfc/V3M/pfc_init_v3m.c
+++ b/drivers/renesas/rcar/pfc/V3M/pfc_init_v3m.c
@@ -676,7 +676,7 @@ void pfc_init_v3m(void)
 				   | IOCTRL30_POC_DU_DR3
 				   | IOCTRL30_POC_DU_DR2);
 
-	pfc_reg_write(PFC_IOCTRL31, IOCTRL31_POC_DUMMY_31
+	pfc_reg_write(PFC_POCCTRL1, IOCTRL31_POC_DUMMY_31
 				   | IOCTRL31_POC_DUMMY_30
 				   | IOCTRL31_POC_DUMMY_29
 				   | IOCTRL31_POC_DUMMY_28
diff --git a/drivers/renesas/rcar/pfc/pfc_regs.h b/drivers/renesas/rcar/pfc/pfc_regs.h
index 4187733..f38cc49 100644
--- a/drivers/renesas/rcar/pfc/pfc_regs.h
+++ b/drivers/renesas/rcar/pfc/pfc_regs.h
@@ -204,8 +204,9 @@
 #define PFC_DRVCTRL23		(PFC_BASE + 0x035CU)
 #define PFC_DRVCTRL24		(PFC_BASE + 0x0360U)
 #define PFC_POCCTRL0		(PFC_BASE + 0x0380U)
-#define PFC_IOCTRL31		(PFC_BASE + 0x0384U)
+#define PFC_POCCTRL1		(PFC_BASE + 0x0384U)
 #define PFC_POCCTRL2		(PFC_BASE + 0x0388U)
+#define PFC_POCCTRL3		(PFC_BASE + 0x038CU)
 #define PFC_TDSELCTRL0		(PFC_BASE + 0x03C0U)
 #define PFC_IOCTRL		(PFC_BASE + 0x03E0U)
 #define PFC_TSREG		(PFC_BASE + 0x03E4U)
-- 
2.7.4

